app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,73 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def transform
+
+      if (params[:date_coverage_1_year].length>0)
+        yyyy1= params[:date_coverage_1_year].to_i 
+      else
+        yyyy2 = 0 #error 
+      end  
+      if (!params[:date_coverage_1_month].eql?("--"))
```

Comment:
== is more of C and Java style, Sufia code uses eql? instead of ==, I looked at the link you shared, that also talks about using eql when comparing instead of ==